pkg/diagnostics/types/diagnostic.go

```diff
@@ -0,0 +1,175 @@
+package types
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"runtime"
+	"strings"
+
+	"github.com/openshift/origin/pkg/diagnostics/log"
+)
+
+type Diagnostic interface {
+	Name() string
+	Description() string
+	CanRun() (canRun bool, reason error)
+	Check() *DiagnosticResult
+}
+
+type DiagnosticResult struct {
```

Comment:
External interaction is through methods.  I don't object to that, but if that's so, then you should the types should reflect that.  This is rightly a public interface with a private impl.